src/main/java/com/puppycrawl/tools/checkstyle/JavadocPropertiesGenerator.java

```diff
@@ -103,36 +101,42 @@ public static void main(String... args)
+     * @noinspection IOResourceOpenedButNotSafelyClosed
      */
     private static void writePropertiesFile(String inputFile, String outputFile)
-            throws CheckstyleException, IOException {
-        final FileOutputStream fos = new FileOutputStream(outputFile);
-        final Writer osw = new OutputStreamWriter(fos, StandardCharsets.UTF_8);
-        final PrintWriter writer = new PrintWriter(osw, false);
-
+            throws CheckstyleException {
+        PrintWriter writer = null;
         try {
-            for (DetailAST top = JavaParser.parseFile(new File(inputFile),
-                    JavaParser.Options.WITH_COMMENTS); top != null; top = top.getNextSibling()) {
-                if (top.getType() != TokenTypes.CLASS_DEF) {
-                    continue;
-                }
-                // OBJBLOCK is always the last child of CLASS_DEF.
-                final DetailAST objBlock = top.getLastChild();
-                for (DetailAST ast = objBlock.getFirstChild(); ast != null;
-                        ast = ast.getNextSibling()) {
-                    if (isPublicStaticFinalIntField(ast)) {
-                        final DetailAST modifiers = ast.findFirstToken(TokenTypes.MODIFIERS);
-                        final String firstJavadocSentence = getFirstJavadocSentence(modifiers);
-                        if (firstJavadocSentence != null) {
-                            writer.println(getName(ast) + "=" + firstJavadocSentence.trim());
-                        }
+            final FileOutputStream fos = new FileOutputStream(outputFile);
+            final Writer osw = new OutputStreamWriter(fos, StandardCharsets.UTF_8);
+            writer = new PrintWriter(osw, false);
+            DetailAST top = JavaParser.parseFile(new File(inputFile),
+                JavaParser.Options.WITH_COMMENTS);
+
+            // Skip the package definition and all imports.
+            while (top.getType() != TokenTypes.CLASS_DEF) {
+                top = top.getNextSibling();
+            }
+
+            // Get the class body and iterate over all its fields.
+            final DetailAST objBlock = top.findFirstToken(TokenTypes.OBJBLOCK);
+            for (DetailAST member = objBlock.getFirstChild(); member != null;
+                 member = member.getNextSibling()) {
+                if (isPublicStaticFinalIntField(member)) {
+                    final DetailAST modifiers = member.findFirstToken(TokenTypes.MODIFIERS);
+                    final String firstJavadocSentence = getFirstJavadocSentence(modifiers);
+                    if (firstJavadocSentence != null) {
+                        writer.println(getName(member) + "=" + firstJavadocSentence.trim());
                     }
                 }
             }
         }
+        catch (IOException ex) {
+            throw new CheckstyleException("Failed to write javadoc properties of '" + inputFile
+                + "' to '" + outputFile + "'", ex);
```

Comment:
Please remove input file reference, it is completely unrelated.
In case of problems with input file , there will be CheckstyleException , you do not catch it , so it is not your context

Reply:
done

Reply:
Actually, `JavaParser.parseFile` throws both `IOException` and `CheckstyleException`.
In case of a missing file a `FileNotFoundException` will be thrown.